nautobot/extras/models/change_logging.py

```diff
@@ -181,3 +181,68 @@ def to_csv(self):
 
     def get_action_class(self):
         return ObjectChangeActionChoices.CSS_CLASSES.get(self.action)
+
+    def get_next_change(self, user=None):
+        """Return next change for this changed object, optionally restricting by user view permission"""
+        related_changes = self.get_related_changes().restrict(user, "view")
+        if user is not None:
+            related_changes = related_changes.restrict(user, "view")
+        return related_changes.filter(time__gt=self.time).order_by("time").first()
+
+    def get_prev_change(self, user=None):
+        """Return previous change for this changed object, optionally restricting by user view permission"""
+        related_changes = self.get_related_changes()
+        if user is not None:
+            related_changes = related_changes.restrict(user, "view")
+        return related_changes.filter(time__lt=self.time).order_by("-time").first()
```

Comment:
These already exist by default on date/datetime fields. You'll have `get_next_by_time()` and `get_previous_by_time()` automatically. Those methods do accept `**kwargs` to be able to do custom filtering. Ideally those would be acceptable.

Reply:
Unfortunately those methods don't support restricting on user permissions so it looks like this is our only option for now.

Reply:
Cool thanks for trying.